lib/configurators/databases/aws.go

```diff
@@ -0,0 +1,656 @@
+func (a *awsConfigurator) build() error {
+	var err error
+
+	// When running the command in manual mode, we want to have zero dependency
+	// with AWS configurations (like awscli or environment variables), so that
+	// the user can run this command and generate the instructions without any
+	// pre-requisite.
+	if !a.flags.Manual {
+		stsClient, err := a.getAWSSTSClient()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		a.awsIdentity, err = awslib.GetIdentityWithClient(context.Background(), stsClient)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	// Define the target and target type.
+	a.target, a.targetType, err = policiesTarget(a.flags, a.awsIdentity)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Generate policies.
+	a.policyDocument, a.policyBoundaryDocument, err = buildPolicyDocuments(
+		a.targetType,
+		a.flags,
+		a.fileConfig,
+		a.awsIdentity,
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Generate formatted policies.
+	a.formattedPolicyDocument, err = formatPolicyDocument(a.policyDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	a.formattedPolicyBoundaryDocument, err = formatPolicyDocument(a.policyBoundaryDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// retrieveCurrentPolicy returns information from AWS about a policy. If the
+// policy doesn't have `teleport` tag and error will be returned.
+func (a *awsConfigurator) retrieveCurrentPolicy(policyName string) (string, []*iam.PolicyVersion, error) {
+	if a.awsIdentity == nil {
+		return "", nil, trace.NotFound("AWS identity not provided, unable to determine policy account ID")
+	}
+
+	policyArn := fmt.Sprintf("arn:aws:iam::%s:policy/%s", a.awsIdentity.GetAccountID(), policyName)
+	client, err := a.getAWSIAMClient()
+	if err != nil {
+		return "", nil, trace.Wrap(err)
+	}
+
+	getPolicyResp, err := client.GetPolicy(&iam.GetPolicyInput{PolicyArn: aws.String(policyArn)})
+	if err != nil {
+		return policyArn, nil, wrapAWSError(err)
+	}
+
+	// Check if the policy has Teleport tag.
+	if !hasTeleportTag(getPolicyResp.Policy.Tags) {
+		return policyArn, nil, trace.BadParameter("Policy %q already exists and was not created by Teleport configurator.", policyName)
+	}
+
+	resp, err := client.ListPolicyVersions(&iam.ListPolicyVersionsInput{PolicyArn: aws.String(policyArn)})
+	if err != nil {
+		return policyArn, nil, wrapAWSError(err)
+	}
+
+	return policyArn, resp.Versions, nil
+}
+
+// upsertPolicyDocument creates a new Policy or creates a Policy version if a
+// policy with the same name already exists.
+//
+// Since policies can hold only five versions, we need to delete a policy
+// version (if the limit is reached) and create a new version.
+// Ref: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-versioning.html#version-limits
+func (a *awsConfigurator) upsertPolicyDocument(documentName, documentDescription string, document *awslib.PolicyDocument) (string, error) {
```

Comment:
Feels like this method should not be a part of configurator but rather some helper method that accepts AWS client and other parameters or maybe some a method of AWSClient object that wraps AWS clients and provides helper methods like this one.